### PR TITLE
Fix `npm run native ios` with xcode12 Command Line Tools

### LINKED ISSUE #2651: Failure building the iOS demo app from the command line
The build fails when trying to build the iOS demo app from the command line.

Steps to reproduce:
1. Do a typical `npm ci` in the gb-mobile root to install the NPM packages from scratch
2. In a second terminal window, run `npm run start:reset` to launch Metro
3. In the first terminal window, do `npm run core ios` to build and launch the iOS demo app in a simulator
4. The build will generally go a long way, the simulator will open but, in the stage where the app would be installed the following error occurs:

```
info Found Xcode workspace "gutenberg.xcworkspace"
info Building (using "xcodebuild -workspace gutenberg.xcworkspace -configuration Debug -scheme gutenberg -destination id=860DC2DE-1126-4395-98E2-B5A89E804633 -derivedDataPath build/gutenberg")
...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
info Installing "DerivedData/Build/Products/Debug-iphonesimulator/gutenberg.app"
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=2):
Failed to install the requested application
An application bundle was not found at the provided path.
Provide a valid path to the desired application bundle.
Print: Entry, ":CFBundleIdentifier", Does Not Exist
error Command failed: /usr/libexec/PlistBuddy -c Print:CFBundleIdentifier DerivedData/Build/Products/Debug-iphonesimulator/gutenberg.app/Info.plist
Print: Entry, ":CFBundleIdentifier", Does Not Exist
. Run CLI with --verbose flag for more details.
Error: Command failed: /usr/libexec/PlistBuddy -c Print:CFBundleIdentifier DerivedData/Build/Products/Debug-iphonesimulator/gutenberg.app/Info.plist
Print: Entry, ":CFBundleIdentifier", Does Not Exist

    at checkExecSyncError (child_process.js:630:11)
    at Object.execFileSync (child_process.js:648:15)
    at runOnSimulator (/Users/stefanos/proj/a8c/rn/t1/gutenberg-mobile/gutenberg/node_modules/@react-native-community/cli-platform-ios/build/commands/runIOS/index.js:183:45)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Command.handleAction (/Users/stefanos/proj/a8c/rn/t1/gutenberg-mobile/gutenberg/node_modules/react-native/node_modules/@react-native-community/cli/build/index.js:182:9)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @wordpress/react-native-editor@1.37.0 ios: `react-native run-ios`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @wordpress/react-native-editor@1.37.0 ios script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stefanos/.npm/_logs/2020-09-21T10_53_26_741Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gutenberg@9.0.0 native: `npm run --prefix packages/react-native-editor "ios"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gutenberg@9.0.0 native script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stefanos/.npm/_logs/2020-09-21T10_53_26_781Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gutenberg-mobile@1.36.0 core: `cd gutenberg && npm run native "ios"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gutenberg-mobile@1.36.0 core script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stefanos/.npm/_logs/2020-09-21T10_53_26_808Z-debug.log
```
5. Error doesn't happen if you open the demo project on Xcode directly and build+run from there

On macos 10.15.6, Xcode v12.0

### PR DESCRIPTION
**Pull request by etoledom**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2651

## Description

This PR fixes the error:
`Print: Entry, ":CFBundleIdentifier", Does Not Exist`
when running `npm run native ios` with xcode12 Command Line Tools.

The script expects the product to be called `gutenberg.app`, but on `project.pbxproj` it was specified to be `GutenbergDemo.app`.
Not sure how this was working before, maybe a change on Xcode12 made this error happen.

This is open for better solutions.

## How has this been tested?

- Run `npm run native ios`.
- Check that the app builds and run as expected.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/react-native-editor/package.json [hunk 1]
```diff
@@ -114,7 +114,7 @@
...
     "test:e2e:build-app:android": "npm run test:e2e:bundle:android && cd android && ./gradlew clean && ./gradlew assembleDebug",
     "test:e2e:install-app:android": "cd android && ./gradlew installDebug",
-    "test:e2e:bundle:ios": "mkdir -p ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app && npm run rn-bundle -- --reset-cache --platform=ios --dev=false --minify false --entry-file=index.js --bundle-output=./ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app/main.jsbundle --assets-dest=./ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app",
+    "test:e2e:bundle:ios": "mkdir -p ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app && npm run rn-bundle -- --reset-cache --platform=ios --dev=false --minify false --entry-file=index.js --bundle-output=./ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app/main.jsbundle --assets-dest=./ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app",
```
Comment (ceyhun) [inline packages/react-native-editor/package.json:117]: I think we might need the same change in gutenberg-mobile/package.json as well

**Review by ceyhun — COMMENTED** [review]
Tried both with Xcode 11 & 12 Command Line Tools and works great! We may need a gutenberg-mobile PR as well though.

File: packages/react-native-editor/package.json (hunk 1 above)
Comment (etoledom) [reply packages/react-native-editor/package.json:117]: True that!
Made this change here: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2658

Good catch 👍

**Review by ceyhun — APPROVED** [review]
🚢 Will just trigger CI again to see if iOS build caching works

**Pull request merged**